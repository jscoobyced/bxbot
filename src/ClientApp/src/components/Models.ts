export class Constants {
    public static readonly StaticContent = (process.env.mode === 'development') ? '' : '/dist';

    public static readonly ImagePath = `${Constants.StaticContent}/images/`;
}

export interface SelectOption {
    value: number;
    text: string;
}
