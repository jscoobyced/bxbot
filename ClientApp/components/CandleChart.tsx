import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import { GoogleCharts } from 'google-charts';
import { DateUtil } from '../utils/date';

interface FetchDataState {
    pairings: Pairing[];
    loading: boolean;
}

export class CandleChart extends React.Component<RouteComponentProps<{}>, FetchDataState> {
    private static ChartData: Array<Pairing>;

    constructor() {
        super();
        this.state = { pairings: [], loading: true };

        fetch('api/Data/pairing/1/5')
            .then(response => response.json() as Promise<Pairing[]>)
            .then(data => {
                CandleChart.ChartData = data;
                this.setState({ loading: false });
            });
    }

    public render() {
        let contents: JSX.Element = <span></span>;
        if (this.state.loading) {
            contents = <p><em>Loading...</em></p>
        }
        else {
            GoogleCharts.load(this.drawChart);
        }

        return <div>
            <h1>BTC on bx.in.th</h1>
            <div id="chart_div"></div>
            {contents}
        </div>;
    }

    private drawChart() {
        const rawData: Array<Pairing> = CandleChart.ChartData;
        const formatedData = new Array<any>();
        const lengthToShow = 96;
        const startIndex = Math.max(0, rawData.length - lengthToShow);
        for (let i = startIndex; i < rawData.length; i++) {
            formatedData.push([
                DateUtil.toHumanHours(new Date(rawData[i].timestamp - 7 * 60 * 60 * 1000)),
                rawData[i].low,
                rawData[i].open,
                rawData[i].close,
                rawData[i].high
            ]);
        }
        var data = GoogleCharts.api.visualization.arrayToDataTable(formatedData, true);

        var options = {
            legend: 'none',
            height: 500,
            candlestick: {
                fallingColor: { strokeWidth: 0, fill: '#a52714' },
                risingColor: { strokeWidth: 0, fill: '#0f9d58' },
                hollowIsRising: true
            },
            colors: ['black']
        };

        var chart = new GoogleCharts.api.visualization.CandlestickChart(document.getElementById('chart_div'));

        chart.draw(data, options);
    }
}

interface Pairing {
    timestamp: number;
    low: number;
    high: number;
    close: number;
    open: number;
}