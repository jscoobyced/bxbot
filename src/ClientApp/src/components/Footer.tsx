import * as React from 'react';

const version = process.env.VERSION;

export class Footer extends React.Component {
    public render() {
        return <footer>${version}</footer>;
    }
}
