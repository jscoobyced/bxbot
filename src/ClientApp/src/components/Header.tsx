import * as React from 'react';
import { NavLink } from 'react-router-dom';
import { ContentKey } from '../services/ContentKey';
import { ContentService } from '../services/ContentService';

export class Header extends React.Component {

    public render() {
        const websiteName = ContentService.getContent(ContentKey.WebsiteName);
        return <header>
            <img src='/images/wings.svg' className='logo' alt={websiteName + ' logo'} title={websiteName} />
            <span>{websiteName}</span>
            <nav>
                <NavLink to={'/'}>
                    Home
                </NavLink>
                <a href='#' className='menu-space'>&nbsp;</a>
                <a href='#' className='dropdown'>Features</a>
                <div className="dropdown-content">
                <NavLink to={'/candle'}>
                        CandleChart
                    </NavLink>
                    <NavLink to={'/'}>
                        Stuff #1
                    </NavLink>
                    <NavLink to={'/candle'}>
                        Stuff #2
                    </NavLink>
                </div>
                <NavLink to={'/'}>
                    About
                </NavLink>
            </nav>
        </header>;
    }
}