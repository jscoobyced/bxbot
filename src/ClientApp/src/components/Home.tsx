import * as React from 'react';

export class Home extends React.Component {
    public render() {
        return <article>
            <h1>Hello, world!</h1>
        </article>;
    }
}
