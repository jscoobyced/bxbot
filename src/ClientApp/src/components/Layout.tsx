import * as React from 'react';
import { Link, NavLink } from 'react-router-dom';

export interface LayoutProps {
    children?: React.ReactNode;
}

export class Layout extends React.Component<LayoutProps, {}> {
    public render() {
        return <div>
            <div>
                <NavLink to={'/'}>
                    Home
                </NavLink> |&nbsp;
                <NavLink to={'/candle'}>
                    CandleChart
                </NavLink>
            </div>
            {this.props.children}
        </div>;
    }
}
