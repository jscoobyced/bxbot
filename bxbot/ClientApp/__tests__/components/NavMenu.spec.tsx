import * as React from 'react';
import { shallow, mount, render } from 'enzyme';
import { NavMenu } from '../../components/NavMenu';
import { } from 'jest';

describe('Layout component', () => {
    it('should render without throwing an error', () => {
        expect(shallow(<NavMenu />)
            .exists())
            .toBe(true)
    })

    it('contains a button for screen reader', () => {
        expect(shallow(<NavMenu />)
            .find('button .sr-only').length)
            .toEqual(1)
    })

    it('contains 2 links', () => {
        expect(shallow(<NavMenu />)
            .find('.navbar-nav li').length)
            .toEqual(2)
    })
});