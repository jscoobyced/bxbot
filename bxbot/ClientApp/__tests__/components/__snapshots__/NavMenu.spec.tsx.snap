// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot 1`] = `
<div
  className="main-nav"
>
  <div
    className="navbar navbar-inverse"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle"
        data-target=".navbar-collapse"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
      <Link
        className="navbar-brand"
        replace={false}
        to="/"
      >
        bxbot
      </Link>
    </div>
    <div
      className="clearfix"
    />
    <div
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav"
      >
        <li>
          <NavLink
            activeClassName="active"
            exact={true}
            to="/"
          >
            <span
              className="glyphicon glyphicon-home"
            />
             Home
          </NavLink>
        </li>
        <li>
          <NavLink
            activeClassName="active"
            to="/candle"
          >
            <span
              className="glyphicon glyphicon-th-list"
            />
             CandleChart
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
